CMakePresets.json

```diff
@@ -0,0 +1,173 @@
+      },
+      "environment": {
+        "LOG_DIR": "${sourceDir}/build/${presetName}/logs"
+      },
+      "hidden": true
+    },
+    {
+      "name": "release",
+      "displayName": "Release",
+      "description": "Same as RelWithDebInfo, but disables debug information",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Release"
+      },
+      "inherits": [
+        "base"
+      ]
+    },
+    {
+      "name": "relwithdebinfo",
+      "displayName": "RelWithDebInfo",
+      "description": "Enables optimizations (-Og or -O2) with debug information",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "RelWithDebInfo"
+      },
+      "inherits": [
+        "base"
+      ]
+    },
+    {
+      "name": "debug",
+      "displayName": "Debug",
+      "description": "Disables optimizations (-O0), enables debug information",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Debug"
+      },
+      "inherits": [
+        "base"
+      ]
+    },
+    {
+      "name": "debug-clang",
+      "displayName": "Debug (w/ clang toolchain)",
+      "environment": {
+        "CC": "clang",
+        "CXX": "clang++"
+      },
+      "inherits": [
+        "debug"
+      ]
+    },
+    {
+      "name": "asan",
+      "displayName": "Debug (w/ ASAN/UBSAN support)",
+      "cacheVariables": {
+        "CLANG_ASAN_UBSAN": "ON"
+      },
+      "environment": {
+        "ASAN_OPTIONS": "'detect_leaks=0;log_path=$env{LOG_DIR}'",
+        "UBSAN_OPTIONS": "'print_stacktrace=1'"
+      },
+      "inherits": [
+        "debug-clang"
+      ]
+    },
+    {
+      "name": "tsan",
+      "displayName": "Debug (w/ TSAN support)",
+      "cacheVariables": {
+        "CLANG_TSAN": "ON"
+      },
+      "environment": {
+        "TSAN_OPTIONS": "'log_path=$env{LOG_DIR}'"
+      },
+      "inherits": [
+        "debug-clang"
+      ]
+    },
+    {
+      "name": "msan",
+      "displayName": "Debug (w/ MSAN support)",
+      "cacheVariables": {
+        "CLANG_MSAN": "ON"
+      },
+      "environment": {
+        "MSAN_OPTIONS": "'log_path=$env{LOG_DIR}'"
+      },
+      "inherits": [
+        "debug-clang"
+      ]
+    },
+    {
+      "name": "win-x64",
+      "displayName": "Windows MSVC x64 (/w VS2019)",
+      "generator": "Visual Studio 17 2022",
+      "architecture": {
+        "value": "x64",
+        "strategy": "set"
+      },
+      "cacheVariables":{
+        "CMAKE_BUILD_TYPE": "RelWithDebInfo"
+      },
+      "inherits": "base",
+      "condition": {
+        "type": "equals",
+        "lhs": "${hostSystemName}",
+        "rhs": "Windows"
+      }
+    },
+    {
+      "name": "win-ninja-x64",
+      "displayName": "Windows x64 (/w Ninja)",
+      "generator": "Ninja",
+      "cacheVariables":{
+        "CMAKE_BUILD_TYPE": "RelWithDebInfo"
+      },
+      "inherits": "base",
+      "condition": {
+        "type": "equals",
+        "lhs": "${hostSystemName}",
+        "rhs": "Windows"
+      }
+    }
+  ],
+  "buildPresets": [
```

Comment:
I don't understand the point of most of these build presets since they don't add any additional configuration compared to a standard `cmake --build build`.

Reply:
Indeed, this is very specific to MSBuild, as that's the only way to make it build in parallel, whereas Ninja, by design, barely has any runtime flags other than `-v`, so I'm not sure how to best handle this tbh 🥲

Reply:
MSBuild only supports project-level parallelism if I recall correctly, so it doesn't really benefit us at all. I think this build presets can safely be removed.

Reply:
good catch! I'll remove it from the defaults then, as it can always be (re-)added in `CMakeUserPresets.json`